csrc/abstract_tensor.cpp

```diff
@@ -413,4 +413,41 @@ std::vector<AbstractTensor> AbstractTensor::unzip() const {
   return result;
 }
 
+namespace {
+
+struct DispatchParallelize {
+  template <typename INPUT>
+  void operator()(ParallelType parallel_type, INPUT&& in) const {
+    using IN = std::decay_t<INPUT>;
+    if constexpr (std::is_same_v<IN, std::monostate>) {
+      return;
+    } else if constexpr (std::is_same_v<IN, IterDomain*>) {
+      return in->parallelize(parallel_type);
+    } else if constexpr (std::is_same_v<IN, ValGroupAndItsGraph>) {
+      // TODO: instead of throwing an error, should we just parallelize all IDs
```

Comment:
Would it make more sense to consider this as the responsibility of the user? I suppose that when using `AbstractTensor`, you'd be aware that each member of the abstract tensor was a ValGroup, so parallelization of the abstract tensor would result in parallelizing all IDs in the group seems to be an intuitive result.

Reply:
Sounds reasonable.